frontend/src/employee-mobile-frontend/components/attendances/child-info/AutoPinLogout.tsx

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useEffect } from 'react'
+
+export default function useAutoLogout(timeout: number, onTimeOut: () => void) {
+  const logout = useCallback(() => {
+    clearTimeout()
+    onTimeOut && onTimeOut()
+  }, [onTimeOut])
```

Comment:
Abstraction level seems inconsistent. This takes onTimeOut callback which could be anything, but part of the naming implies that this hook is logout specific. I guess you can go either direction, but if the logout logic has to be passed in anyway and nothing inside this is really logout specific, I would probably remove mentions of logout and make this a generic useTimeOut or useInactivity hook.

Reply:
Renamed to useInactivityTimeout